pyaedt/modules/MeshIcepak.py

```diff
@@ -1280,10 +1283,94 @@ def automatic_mesh_3D(self, accuracy, enable_stair_step=True):
+
+        Examples
+        --------
+
+        >>> ipk.mesh.assign_priorities([["Box1", "Rectangle1"], ["Box2", "Fan1_1"], ["Heatsink1_1"]])
+        """
+        if not isinstance(assignment[0], list):
+            raise AttributeError("``assignment`` input must be a list of lists.")
+        props = {"PriorityListParameters": []}
+        for level, objects in enumerate(assignment):
+            level += 1
+            if isinstance(objects[0], str):
+                objects = [
+                    self.modeler.objects_by_name.get(o, self.modeler.user_defined_components.get(o, None))
+                    for o in objects
+                ]
+            obj_3d = [
+                o
+                for o in objects
+                if (isinstance(o, Object3d) and o.is3d)
+                or (isinstance(o, UserDefinedComponent) and any(p.is3d for p in o.parts.values()))
+            ]
+            obj_2d = [
+                o
+                for o in objects
+                if (isinstance(o, Object3d) and not o.is3d)
+                or (isinstance(o, UserDefinedComponent) and any(not p.is3d for p in o.parts.values()))
+            ]
+            if obj_3d:
+                level_3d = {
+                    "EntityList": ", ".join([o.name for o in obj_3d]),
+                    "PriorityNumber": level,
+                    "PriorityListType": "3D",
+                }
+                if all(isinstance(o, Object3d) for o in obj_3d):
+                    level_3d["EntityType"] = "Object"
+                elif all(isinstance(o, UserDefinedComponent) for o in obj_3d):
+                    level_3d["EntityType"] = "Component"
+                else:
+                    raise AttributeError("Cannot assign components and parts on the same level.")
+                props["PriorityListParameters"].append(level_3d)
+            if obj_2d:
+                level_2d = {
+                    "EntityList": ", ".join([o.name for o in obj_2d]),
+                    "PriorityNumber": level,
+                    "PriorityListType": "2D",
+                }
+                if all(isinstance(o, Object3d) for o in obj_2d):
+                    level_2d["EntityType"] = "Object"
+                elif all(isinstance(o, UserDefinedComponent) for o in obj_2d):
+                    level_2d["EntityType"] = "Component"
+                else:
+                    raise AttributeError("Cannot assign components and parts on the same level.")
+                props["PriorityListParameters"].append(level_2d)
+        props = {"UpdatePriorityListData": props}
+        args = []
+        _dict2arg(props, args)
+        self.modeler.oeditor.UpdatePriorityList(args[0])
+        return True
+
     @pyaedt_function_handler(obj_list="assignment", comp_name="component")
     def add_priority(self, entity_type, assignment=None, component=None, priority=3):
         """Add priority to objects.
 
+        .. deprecated:: 0.9.1
+        Use :func:`assign_priorities` method instead.
```

Comment:
```suggestion
        Use :func:`assign_priorities` function instead.
```
Assuming this is a function based on the role used.